src/6model/reprs/P6opaque.c

```diff
@@ -1186,12 +1186,12 @@ static void deserialize_repr_data(MVMThreadContext *tc, MVMSTable *st, MVMSerial
     num_classes = (MVMuint16)MVM_serialization_read_int(tc, reader);
     repr_data->name_to_index_mapping = (MVMP6opaqueNameMap *)MVM_malloc((num_classes + 1) * sizeof(MVMP6opaqueNameMap));
     for (i = 0; i < num_classes; i++) {
-        MVMint32 num_attrs = 0;
+        MVMuint16 num_attrs = 0;
 
         MVM_ASSIGN_REF(tc, &(st->header), repr_data->name_to_index_mapping[i].class_key,
             MVM_serialization_read_ref(tc, reader));
 
-        num_attrs = MVM_serialization_read_int(tc, reader);
+        num_attrs = (MVMuint16)MVM_serialization_read_int(tc, reader);
         repr_data->name_to_index_mapping[i].names = (MVMString **)MVM_malloc(P6OMAX(num_attrs, 1) * sizeof(MVMString *));
```

Comment:
I think that these two are wrong. Instead `j` should be MVMuint32. ie split:

    MVMuint16 i, j, num_classes;

Reply:
Yep, fixed and force pushed.